### PR TITLE
[CARBONDATA-3889] Cleanup typo code for carbondata-core module

### PR DESCRIPTION
**Pull request by QiangCai**
### Why is this PR needed?
 There are many typos in carbondata-core module
 
 ### What changes were proposed in this PR?
Cleanup typo code for carbondata-core module
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3318/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1579/

**Comment by Indhumathi27** [comment]
Please Clean code for DictionaryByteArrayWrapper, Cache

File: core/src/main/java/org/apache/carbondata/core/cache/Cacheable.java [hunk 1]
```diff
@@ -40,7 +40,7 @@
 
   /**
    * Method to be used for invalidating the cacheable object. API to be invoked at the time of
-   * removing the cacheable object from memory. Example at the of removing the cachebale object
+   * removing the cacheable object from memory. Example at removing the cacheable object
```
Comment (Indhumathi27) [inline core/src/main/java/org/apache/carbondata/core/cache/Cacheable.java:43]: Remove return tag in this interface

Comment (QiangCai) [reply core/src/main/java/org/apache/carbondata/core/cache/Cacheable.java:43]: I will do it in new PR. This PR focus on the typo.

**Comment by QiangCai** [comment]
@Indhumathi27 
I will do it in new PR. This PR focus on the typo.

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3419/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1677/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/3426/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1684/

**Comment by kevinjmh** [comment]
LGTM

**Pull request closed without merging**